File: gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/model/StoreType.java
/*
 * Copyright 2020 Crown Copyright
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package uk.gov.gchq.gaffer.gaas.model;

import com.fasterxml.jackson.annotation.JsonValue;

public enum StoreType {

    ACCUMULO_STORE("accumuloStore"),
    FEDERATED_STORE("federatedStore"),
    MAP_STORE("mapStore"),
    PROXY_STORE("proxyStore");

    private final String value;

    StoreType(final String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }
}

File: gaas-ui/src/domain/store-type.ts
export enum StoreType {
    OTHER = "O",
    FEDERATED_STORE  = "F",
    ACCUMULO = "A",
    MAPSTORE = "M",
    PROXY_STORE="P"
}

export function getStoreType(storeType: string): StoreType {

    switch (storeType) {
        case "accumuloStore":
            return StoreType.OTHER;
        case "mapStore":
            return StoreType.OTHER;
        case "federatedStore":
            return StoreType.FEDERATED_STORE;
        case "proxyStore":
            return StoreType.OTHER;
        default:
             throw new Error(storeType+" is not a supported store type");

    }
}
